### PR TITLE
Index Sets UI

### PR DESCRIPTION
**Pull request by bernd**
This PR creates UI components for index sets and moves the existing list of indices into the index set details.

image

image

image

### CODE REVIEW

**Comment by bernd** [comment]
Integration tests will be fixed with #3133

**Comment by bernd** [comment]
Rebased against master and added some smaller details.

I will not touch it anymore now. Happy review. :smiley:

File: graylog2-web-interface/src/stores/indices/DeflectorStore.js [hunk 1]
```diff
@@ -12,23 +12,20 @@ const DeflectorStore = Reflux.createStore({
...
-  init() {
-    DeflectorActions.list();
-  },
   getInitialState() {
     return { deflector: this.deflector };
   },
-  cycle() {
-    const url = URLUtils.qualifyUrl(ApiRoutes.DeflectorApiController.cycle().url);
+  cycle(indexSetId) {
+    const url = URLUtils.qualifyUrl(ApiRoutes.DeflectorApiController.cycle(indexSetId).url);
```
Comment (kroepke) [inline graylog2-web-interface/src/stores/indices/DeflectorStore.js]: I believe this still goes to the `/cluster/deflector` endpoint:
```
FetchProvider.js:55 There was an error fetching a resource: cannot POST http://localhost:9000/api/cluster/deflector/5832e27e15dd0cf04ea017df/cycle (404). Additional information: HTTP 404 Not Found
```

File: graylog2-web-interface/src/routing/ApiRoutes.js [hunk 2]
```diff
@@ -60,8 +61,8 @@ const ApiRoutes = {
     update: (decoratorId) => { return { url: `/search/decorators/${decoratorId}` }; },
   },
   DeflectorApiController: {
-    cycle: () => { return { url: '/cluster/deflector/cycle' }; },
-    list: () => { return { url: '/system/deflector' }; },
+    cycle: (indexSetId) => { return { url: `/cluster/deflector/${indexSetId}/cycle` }; },
```
Comment (kroepke) [inline graylog2-web-interface/src/routing/ApiRoutes.js]: `/system/deflector`?

File: graylog2-web-interface/src/pages/IndexSetCreationPage.jsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            <span>
+              <LinkContainer to={Routes.SYSTEM.INDICES.LIST}>
+                <Button bsStyle="info">Overview</Button>
+              </LinkContainer>
+            </span>
+          </PageHeader>
+
+          <Row className="content">
+            <Col md={12}>
+              <IndexSetConfigurationForm indexSet={indexSet}
```
Comment (kroepke) [inline graylog2-web-interface/src/pages/IndexSetCreationPage.jsx]:
```
 Required prop `cancelLink` was not specified in `IndexSetConfigurationForm`. Check the render method of `IndexSetCreationPage`.
```

File: graylog2-web-interface/src/components/indices/IndexSetConfigurationForm.jsx [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+                 required />
+          {readOnlyconfig}
+          <Input wrapperClassName="col-sm-offset-3 col-sm-7">
+            {rotationConfig}
+            {retentionConfig}
+
+            <hr/>
+            <Button type="submit" bsStyle="success">Save</Button>
+            &nbsp;
+            <LinkContainer to={this.props.cancelLink}>
```
Comment (kroepke) [inline graylog2-web-interface/src/components/indices/IndexSetConfigurationForm.jsx]: the other comment leads to:
```
 Required prop `to` was not specified in `LinkContainer`. Check the render method of `IndexSetConfigurationForm`.
```

File: graylog2-web-interface/src/components/indices/IndicesMaintenanceDropdown.jsx [hunk 5]
```diff
@@ -9,22 +9,27 @@ import StoreProvider from 'injection/StoreProvider';
 const DeflectorStore = StoreProvider.getStore('Deflector'); // eslint-disable-line no-unused-vars
 
 const IndicesMaintenanceDropdown = React.createClass({
+  propTypes: {
+    indexSetId: React.PropTypes.string.isRequired,
+  },
+
   _onRecalculateIndexRange() {
     if (window.confirm('This will trigger a background system job. Go on?')) {
+      // TODO 2.2: Only rebuild index set here? It currently rebuilds all index ranges!
```
Comment (kroepke) [inline graylog2-web-interface/src/components/indices/IndicesMaintenanceDropdown.jsx]: For now I would simply change the wording to "all index ranges" and fix this post-beta.
Either by moving the action to the index sets overview page, or by implementing another system job that operates on a single index set.

File: graylog2-web-interface/src/routing/ApiRoutes.js (hunk 2 above)
Comment (bernd) [inline graylog2-web-interface/src/routing/ApiRoutes.js]: No, this needs to be `/cluster/..` to make this work on on-master nodes.

The `/cluster/deflector/{indexSetId}/cycle` endpoint was just missing.

File: graylog2-web-interface/src/stores/indices/DeflectorStore.js (hunk 1 above)
Comment (bernd) [inline graylog2-web-interface/src/stores/indices/DeflectorStore.js]: See below.

File: graylog2-web-interface/src/components/indices/IndexSetConfigurationForm.jsx (hunk 4 above)
Comment (bernd) [inline graylog2-web-interface/src/components/indices/IndexSetConfigurationForm.jsx]: This is because of the issue above. Should be fixed.

**Pull request merged**